app/models/manageiq/providers/hawkular/common/event_catcher/runner_mixin.rb

```diff
@@ -0,0 +1,50 @@
+module ManageIQ::Providers::Hawkular::Common::EventCatcher::RunnerMixin
+  extend ActiveSupport::Concern
+
+  def log_handle
+    self.class.log_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      new_events = events.select { |e| whitelist?(e) }
```

Comment:
Copying comment from #14195

@jshaughn @cfcosta extracted the whitelist / blacklist logic from hawkular, but this might be a little confusing for future readers. I think people usually use a white list (take only what is on this list) or a black list (take every thing except for this list). Also these two happen in different stages (before and after queuing them)

Maybe we should name these differently ?